### PR TITLE
update(query):  StatefulSet Without Service Name for Kubernetes

### PR DESCRIPTION
**Pull request by cxMiguelSilva**
**Proposed Changes**
- Modify check for matched labels between Headless service and StatefulSet

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

**Comment by kaplanlior** [comment]
This PR fixes a problem reported at https://github.com/Checkmarx/kics/discussions/5137

File: assets/queries/k8s/statefulset_without_service_name/query.rego [hunk 1]
```diff
@@ -1,10 +1,18 @@
-package Cx
+package Cx
```
Comment (rafaela-soares) [inline assets/queries/k8s/statefulset_without_service_name/query.rego:1]: Please, also update the query `kics\assets\queries\terraform\kubernetes\statefulset_without_service_name`

If possible, also improve the query description, please

File: assets/queries/k8s/statefulset_without_service_name/query.rego [hunk 2]
```diff
@@ -14,9 +22,17 @@ CxPolicy[result] {
...
+unmatchedLabels(serviceLabels, statefulsetLabels){
+    value := serviceLabels[name]
+    notContain(value, name , statefulsetLabels)
+}
+
+notContain( value, name, list){
+	list[name] != value
+}else{
+	not common_lib.valid_key(list, name)
 }
```
Comment (rafaela-soares) [inline assets/queries/k8s/statefulset_without_service_name/query.rego:38]:
```suggestion
match_labels(serviceLabels, statefulsetLabels) {
    count({x | label := serviceLabels[x]; label == statefulsetLabels[_]}) == count(serviceLabels)
}
```

File: assets/queries/k8s/statefulset_without_service_name/query.rego [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
 	statefulset := input.document[i]
 	statefulset.kind == "StatefulSet"
 
-	count({x | resource := input.document[x]; resource.kind == "Service"; resource.spec.clusterIP == "None"; statefulset.metadata.namespace == resource.metadata.namespace; statefulset.spec.serviceName == resource.metadata.name; labelsMatch(statefulset, resource) == true}) == 0
+	count({x | 	resource := input.document[x];
+    			resource.kind == "Service"; 
+            	resource.spec.clusterIP == "None"; 
+            	statefulset.metadata.namespace == resource.metadata.namespace; 
+            	statefulset.spec.serviceName == resource.metadata.name; 
+            	not unmatchedLabels( resource.spec.selector, statefulset.spec.template.metadata.labels)
```
Comment (rafaela-soares) [inline assets/queries/k8s/statefulset_without_service_name/query.rego:14]:
```suggestion
                match_labels(resource.spec.selector, statefulset.spec.template.metadata.labels)
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/statefulset_without_service_name/query.rego [hunk 4]
```diff
@@ -14,9 +22,10 @@ CxPolicy[result] {
 		"issueType": "IncorrectValue",
 		"keyExpectedValue": sprintf("metadata.name=%s.spec.serviceName refers to a Headless Service", [metadata]),
```
Comment (joaorufi) [inline assets/queries/k8s/statefulset_without_service_name/query.rego:23]:
```suggestion
		"keyExpectedValue": sprintf("metadata.name=%s.spec.serviceName should refer to a Headless Service", [metadata]),
```

Please, also update on the terraform query

**Review by joaorufi — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/statefulset_without_service_name/metadata.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
   "queryName": "StatefulSet Without Service Name",
   "severity": "LOW",
   "category": "Availability",
-  "descriptionText": "Check if the StatefulSets have a headless 'serviceName'",
+  "descriptionText": "StatefulSets should have a headless 'serviceName'. The headless service labels should also be implemented on StatefulSets labels.",
```
Comment (rafaela-soares) [inline assets/queries/k8s/statefulset_without_service_name/metadata.json:6]:
```suggestion
  "descriptionText": "StatefulSets should not have a headless 'serviceName'. The headless service labels should be implemented on StatefulSets labels.",
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

File: assets/queries/terraform/kubernetes/statefulset_without_service_name/metadata.json [hunk 6]
```diff
@@ -3,7 +3,7 @@
   "queryName": "StatefulSet Without Service Name",
   "severity": "LOW",
   "category": "Availability",
-  "descriptionText": "Check if the StatefulSet have a headless 'serviceName'",
+  "descriptionText": "StatefulSets should have a headless 'serviceName'. The headless service labels should also be implemented on StatefulSets labels.",
```
Comment (rafaela-soares) [inline assets/queries/terraform/kubernetes/statefulset_without_service_name/metadata.json:6]:
```suggestion
  "descriptionText": "StatefulSets should not have a headless 'serviceName'. The headless service labels should be implemented on StatefulSets labels.",
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by rafaela-soares — APPROVED** [review]
LGTM 🚀

**Review by joaorufi — APPROVED** [review]

**Pull request merged**